### PR TITLE
:monocle_face: Booking Limit implements

### PR DESCRIPTION
**Pull request by Ewentualnie**

### CODE REVIEW

**Comment by Mo-og** [comment]
|File|Coverage [47.6%]|:green_apple:|
|:-|:-:|:-:|
|[BookingLimitController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/6524a871e957c509c7d18dcd5b4383eba6240761/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingLimitController.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/6524a871e957c509c7d18dcd5b4383eba6240761/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[BookingLimit.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/6524a871e957c509c7d18dcd5b4383eba6240761/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FBookingLimit.java)|55.81%|:green_apple:|
|[UserController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/6524a871e957c509c7d18dcd5b4383eba6240761/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FUserController.java)|54.05%|:green_apple:|
|[WeekEditException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/6524a871e957c509c7d18dcd5b4383eba6240761/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FWeekEditException.java)|45.45%|:green_apple:|
|[BookingController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/6524a871e957c509c7d18dcd5b4383eba6240761/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingController.java)|0%|:x:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/6524a871e957c509c7d18dcd5b4383eba6240761/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|0%|:x:|

|Total Project Coverage|33.85%|:green_apple:|
|:-|:-:|:-:|

**Comment by Mo-og** [comment]
|File|Coverage [93.15%]|:green_apple:|
|:-|:-:|:-:|
|[WeekEditException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FWeekEditException.java)|100%|:green_apple:|
|[BookingLimitController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingLimitController.java)|100%|:green_apple:|
|[BookingController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingController.java)|100%|:green_apple:|
|[UserController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FUserController.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[BookingLimit.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FBookingLimit.java)|72.09%|:green_apple:|
|[BookingLimitResponse.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/0ba5fe8d47236e9eb23048a5d490774d204294ec/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2Fdto%2FBookingLimitResponse.java)|68.29%|:green_apple:|

|Total Project Coverage|91.55%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   *
+   * @param interviewerId interviewer's id
+   * @param weekNum       week number
+   * @return BookingLimit
+   */
+  public BookingLimit getBookingLimit(Long interviewerId, Integer weekNum) {
+    if (!userRepository.existsById(interviewerId)) {
+      throw new UserNotFoundException(interviewerId + "");
+    }
+    return bookingLimitRepository.findByInterviewerIdAndWeekNum(interviewerId, weekNum);
```
Comment (Mo-og) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java:69]: If nothing was found service should first check if interviewer had any limits set before given weekNum. If they had then return that limit or set that limit to current week and then return.

File: src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  /**
+   * Receive BookingLimit of interviewer by weekNum.
+   *
+   * @param interviewerId interviewer's id
+   * @param weekNum       week number
+   * @return BookingLimit
+   */
+  public BookingLimit getBookingLimit(Long interviewerId, Integer weekNum) {
```
Comment (Mo-og) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java:65]: Should be renamed to smth like `getBookingLimitByInterviewerIdAndWeekNum()` or `obtainBookinLimit()` to differ from getters

File: src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    if (bookingLimit != null) {
+      bookingLimit.setBookingLimit(limit);
+    } else {
+      bookingLimit = new BookingLimit(interviewerId, weekNum, limit);
+    }
+
+    return bookingLimitRepository.save(bookingLimit);
+  }
+
+  public List<BookingLimit> getWeekBookingLimits(Integer weekNum) {
```
Comment (Mo-og) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java:54]: Should be renamed to smth like `getBookingLimitsByWeekNum`

File: src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  /**
+   * set bookings limit for interviewer per week.
+   *
+   * @param limit         bookings limit
+   * @param interviewerId interviewer's id
+   * @return Booking limit
+   */
+  public BookingLimit setBookingLimit(Integer limit, Long interviewerId, Integer weekNum) {
```
Comment (Mo-og) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java:36]: Should be renamed to smth like `saveBookingLimit` to differ from setters

File: src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+   * @return BookingLimit
+   */
+  public BookingLimit getBookingLimit(Long interviewerId, Integer weekNum) {
+    if (!userRepository.existsById(interviewerId)) {
+      throw new UserNotFoundException(interviewerId + "");
+    }
+    return bookingLimitRepository.findByInterviewerIdAndWeekNum(interviewerId, weekNum);
+  }
+
+  public BookingLimitResponse getUserWeekBookingLimit(Long interviewerId, Integer weekNum) {
```
Comment (Mo-og) [inline src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java:72]: Should be renamed

File: src/main/java/com/intellias/intellistart/interviewplanning/controllers/BookingLimitController.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      @RequestParam Integer weekNum) {
+    return bookingLimitService.setBookingLimit(bookingLimit, interviewerId, weekNum);
+  }
+
+  @GetMapping("/interviewers/bookingLimits/week")
+  public List<BookingLimit> getWeekBookingLimits(@RequestParam Integer weekNum) {
+    return bookingLimitService.getWeekBookingLimits(weekNum);
+  }
+
+  @GetMapping("/interviewers/bookingLimits/week/user")
```
Comment (Mo-og) [inline src/main/java/com/intellias/intellistart/interviewplanning/controllers/BookingLimitController.java:38]: Url should be `/interviewers/{interviewerId}/bookingLimits/{weekId}`

**Review by Mo-og — CHANGES_REQUESTED** [review]

File: src/main/java/com/intellias/intellistart/interviewplanning/controllers/BookingLimitController.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Ewentualnie) [reply src/main/java/com/intellias/intellistart/interviewplanning/controllers/BookingLimitController.java:38]: got it

**Comment by Mo-og** [comment]
|File|Coverage [94.51%]|:green_apple:|
|:-|:-:|:-:|
|[BookingLimitRequest.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2Fdto%2FBookingLimitRequest.java)|100%|:green_apple:|
|[WeekEditException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FWeekEditException.java)|100%|:green_apple:|
|[BookingController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingController.java)|100%|:green_apple:|
|[UserController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FUserController.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[BookingLimitController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingLimitController.java)|73.91%|:green_apple:|
|[BookingLimit.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f79070b05f1eef7da9db13c5426b462cf12ecd93/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FBookingLimit.java)|72.09%|:green_apple:|

|Total Project Coverage|91.76%|:green_apple:|
|:-|:-:|:-:|

**Comment by Mo-og** [comment]
|File|Coverage [94.51%]|:green_apple:|
|:-|:-:|:-:|
|[BookingLimitRequest.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2Fdto%2FBookingLimitRequest.java)|100%|:green_apple:|
|[WeekEditException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FWeekEditException.java)|100%|:green_apple:|
|[BookingController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingController.java)|100%|:green_apple:|
|[UserController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FUserController.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[BookingLimitController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingLimitController.java)|73.91%|:green_apple:|
|[BookingLimit.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7b65c705bbf930fb0a05a59013e8954af644cf5a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FBookingLimit.java)|72.09%|:green_apple:|

|Total Project Coverage|91.76%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/models/dto/BookingLimitResponse.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.Getter;
+
+/**
+ * DTO to front end response.
+ */
+@Getter
+public class BookingLimitResponse {
+
+  private final Integer limit;
+
```
[the remark this answers is not in the document]
Comment (Ewentualnie) [reply src/main/java/com/intellias/intellistart/interviewplanning/models/dto/BookingLimitResponse.java]: for what?

**Comment by Mo-og** [comment]
|File|Coverage [94.51%]|:green_apple:|
|:-|:-:|:-:|
|[BookingLimitRequest.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2Fdto%2FBookingLimitRequest.java)|100%|:green_apple:|
|[WeekEditException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FWeekEditException.java)|100%|:green_apple:|
|[BookingController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingController.java)|100%|:green_apple:|
|[UserController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FUserController.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[BookingLimitController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingLimitController.java)|73.91%|:green_apple:|
|[BookingLimit.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/f3c92217a159f7918a59eb4111aaab9b4f90638e/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FBookingLimit.java)|72.09%|:green_apple:|

|Total Project Coverage|91.76%|:green_apple:|
|:-|:-:|:-:|

**Comment by Mo-og** [comment]
|File|Coverage [92.25%]|:green_apple:|
|:-|:-:|:-:|
|[ApplicationErrorException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FApplicationErrorException.java)|100%|:green_apple:|
|[WeekEditException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FWeekEditException.java)|100%|:green_apple:|
|[BookingLimitController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingLimitController.java)|100%|:green_apple:|
|[BookingController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FBookingController.java)|100%|:green_apple:|
|[UserController.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontrollers%2FUserController.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[BookingLimit.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2FBookingLimit.java)|72.09%|:green_apple:|
|[BookingLimitRequest.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodels%2Fdto%2FBookingLimitRequest.java)|69.35%|:green_apple:|
|[BookingLimitException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e6b7b503afddcc627a3d0e234b8c083f1795ec5/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FBookingLimitException.java)|0%|:x:|

|Total Project Coverage|90.79%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/models/dto/BookingLimitRequest.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
+package com.intellias.intellistart.interviewplanning.models.dto;
+
+import java.util.Objects;
+import lombok.Getter;
+
+/**
+ * DTO from front end request.
+ */
+@Getter
+public class BookingLimitRequest {
```
Comment (Mo-og) [inline src/main/java/com/intellias/intellistart/interviewplanning/models/dto/BookingLimitRequest.java:10]: Why name is not BookingLimitDto ?

File: src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  /**
+   * set bookings limit for interviewer per week.
+   *
+   * @param limit         bookings limit
+   * @param interviewerId interviewer's id
+   * @return Booking limit
+   */
+  public BookingLimit setBookingLimit(Integer limit, Long interviewerId, Integer weekNum) {
+    if (!userRepository.existsById(interviewerId)) {
+      throw new UserNotFoundException(interviewerId + "");
```
[the remark this answers is not in the document]
Comment (Mo-og) [reply src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java:40]: > ~Replace `interviewerId + ""` with `interviewerId.toString()`~
> 
> Use InterviewerNotFoundException instead

Still there

File: src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java (hunk 1 above)
Comment (Mo-og) [reply src/main/java/com/intellias/intellistart/interviewplanning/services/BookingLimitService.java:74]: Still there, but can be new issue

File: src/test/java/com/intellias/intellistart/interviewplanning/services/BookingLimitServiceTest.java [hunk 10]
```diff
@@ -57,28 +60,27 @@ void testSetBooking() {
...
+        bookingLimitRequest);
     assertEquals(bookingLimit.getId(), newBookingLimit.getId());
     assertEquals(bookingLimit.getInterviewerId(), newBookingLimit.getInterviewerId());
     assertEquals(bookingLimit.getBookingLimit(), newBookingLimit.getBookingLimit());
   }
 
   @Test
   void testSetBookingThrowUserException() {
-    when(userRepository.existsById(notExistingUserId))
+    lenient().when(userRepository.existsById(notExistingUserId))
```
Comment (Mo-og) [inline src/test/java/com/intellias/intellistart/interviewplanning/services/BookingLimitServiceTest.java:72]: Are you sure lentient is needed?

**Review by Mo-og — APPROVED** [review]

**Pull request merged**